Controller/src/controller/regionalVM.go

```diff
@@ -20,48 +20,80 @@ import (
 	"log"
 )
 
+type vmState int
+
+const (
+	inactive state = iota
+	starting
+	idle
+	probing
+	stopped
+)
+
 type regionalVM struct {
 	name      string
 	zone      string
-	cpuMin    string
-	imageName string
-	active    bool
+	state     vmState
+	stateLock sync.Mutex
 	probes    []*ProbeConfig
+	lastPing  time.Time
 }
 
-func newRegionalVM(name string, zone string, cpu string, img string) *regionalVM {
+func newRegionalVM(name string, zone string) *regionalVM {
 	return &regionalVM{name: name, zone: zone, cpuMin: cpu, imageName: img}
 }
 
-func (vm *regionalVM) startVM(cpu string, img string, sa string) error {
-	err := maker.Command("gcloud", "compute", "instances", "create", vm.name, "--zone", vm.zone,
-		"--min-cpu-platform", cpu, "--image", img, "--service-account", sa).Run()
+func (vm *regionalVM) startVM() error {
+
+	//TODO(langenbahn): Edit this command to include startup script
+	/*err := maker.Command("gcloud", "compute", "instances", "create", vm.name, "--zone", vm.zone,
+		"--quiet", "--min-cpu-platform", config.GetMinCpu(), "--image", config.GetDiskImageName(),
+		"--service-account", config.GetAccount().GetServiceAccount()).Run()
 	if err != nil {
 		return err
-	}
-	vm.active = true
+	}*/
+	log.Print("Starting VM")
+	vm.setState(starting)
 	return nil
 }
 
-func (vm *regionalVM) startProbes(acc *AccountInfo) {
-	s := &ProbeConfigs{Probe: vm.probes}
-	p := proto.MarshalTextString(s)
-	a := proto.MarshalTextString(acc)
+func (vm *regionalVM) stopVM() {
+	/*err := maker.Command("gcloud", "compute", "instances", "delete", vm.name, "--zone", vm.zone, "--quiet").Run()
+	if err != nil {
+		log.Printf("stopVM: unable to stop VM %s in zone %s: %v", vm.name, vm.zone, err)
+	}*/
+	log.Print("Stopping VM")
```

Comment:
I would change this message to be "Stopping VM - Not implemented"

Reply:
Addressed in 85540dd